provision/acc_provision/apic_provision.py

```diff
@@ -603,25 +660,57 @@ def update(data, x):
             for rack in self.config["topology"]["rack"]:
                 for leaf in rack["leaf"]:
                     if "local_ip" in leaf and "id" in leaf:
-                        update(data, self.calico_floating_svi(rack["aci_pod_id"], leaf["id"], leaf["local_ip"]))
+                        update(data, self.calico_floating_svi(rack["aci_pod_id"], leaf["id"], leaf["local_ip"], tenant))
                         if len(node_map) == 0 and ("topology/pod-%s/node-%s" % (rack["aci_pod_id"], leaf["id"])) not in node_ids:
-                            update(data, self.add_configured_nodes(rack["aci_pod_id"], leaf["id"]))
+                            update(data, self.add_configured_nodes(rack["aci_pod_id"], leaf["id"], tenant))
             # If the node already has a router_id in external l3out, then use the same router_id
             if len(node_map) != 0:
                 for node_dn, router_id in node_map.items():
-                    update(data, self.add_configured_nodes_with_routerid(rack["aci_pod_id"], node_dn, router_id))
+                    update(data, self.add_configured_nodes_with_routerid(rack["aci_pod_id"], node_dn, router_id, tenant))
                     if node_dn not in node_ids:
                         node_ids.append(node_dn)
                 # Finally add all other nodes which arent added.
                 for rack in self.config["topology"]["rack"]:
                     for leaf in rack["leaf"]:
                         if ("topology/pod-%s/node-%s" % (rack["aci_pod_id"], leaf["id"])) not in node_ids:
-                            update(data, self.add_configured_nodes(rack["aci_pod_id"], leaf["id"]))
+                            update(data, self.add_configured_nodes(rack["aci_pod_id"], leaf["id"], tenant))
 
             update(data, self.l3out_filter())
-            update(data, self.l3out_brcp())
-            update(data, self.ext_epg_svc())
-            update(data, self.ext_epg_int())
+
+            external_l3out_tn = self.config["aci_config"]["vrf"]["tenant"]
+            external_l3out_vrf = self.config["aci_config"]["vrf"]["name"]
+
+            if cluster_l3out_tn != external_l3out_tn and cluster_l3out_vrf != external_l3out_vrf:
+                """
+                # different tenant, different vrf
+                1. Create global scope contract under provider (vrf) tenant
+                2. Export global scope contract under consumer(cluster_l3out) tenant
+                3. Consumer tenant has the floating svi l3 out. The ext epgs under floating
+                   l3out consume the exported contract
+                """
+                update(data, self.l3out_brcp_global())
+                update(data, self.cluster_l3out_cif_global())
+                update(data, self.ext_epg_svc_contract_interface())
+                update(data, self.ext_epg_int_contract_interface())
+            elif cluster_l3out_tn == external_l3out_tn and cluster_l3out_vrf != external_l3out_vrf:
```

Comment:
an additional case for the same block. If tenants are different but external_l3out_tn == common, fall into here.

Reply:
@shastrinator Addressed this one